NebulaPatcher/Patches/Dynamic/GameData_Patch.cs

```diff
@@ -86,4 +87,69 @@ public static bool Prefix(GameData __instance, PlanetData planet)
             return false;
         }
     }
+    // NOTE: this is part of the weird planet movement fix, see ArrivePlanet() patch for more information
+    [HarmonyPatch(typeof(GameData), "OnActivePlanetLoaded")]
+    class GameData_Patch3
+    {
+        public static bool Prefix(GameData __instance, PlanetData planet)
+        {
+            if (LocalPlayer.IsMasterClient)
+            {
+                return true;
+            }
+            if (planet != null)
+            {
+                if (planet.factoryLoaded)
+                {
+                    __instance.OnActivePlanetFactoryLoaded(planet);
+                }
+                else
+                {
+                    planet.LoadFactory();
+                    planet.onFactoryLoaded += __instance.OnActivePlanetFactoryLoaded;
+                }
+            }
+            planet.onLoaded -= __instance.OnActivePlanetLoaded;
```

Comment:
This should be under the null check, otherwise it will NRE.

Reply:
Also, onFactoryLoaded vs onLoaded? Was this meant to be done in the event handler?